tests/testthat/test-test-threads.R

```diff
@@ -67,6 +74,61 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
   })
 
 
+  test_that("test_threads(.dry_run=T) correctly changes maxeval/niter for multiple estimation methods [BBR-TSTT-002]", {
+
+    mods_complex <- test_threads(mod_complex, .threads = c(2, 4), .max_eval = 100, .mode = "local", .dry_run = TRUE)
+
+    search_str <- "MAXEVAL|NITER"
+    max_evals <- map(mods_complex, function(.mod){
+      mod_path <- get_model_path(.mod)
+      mod_lines <- mod_path %>% readLines()
+      str_line_loc <- which(grepl(search_str, mod_lines))
+      str_values <- str_split(mod_lines[str_line_loc], " ")
+
+      values <- map(1:length(str_values), ~ {
+        est_loc <- grepl("METHOD", str_values[[.x]])
```

Comment:
I have two minor comments on the new tests, but I think they're fine as is.

  * `expect_identical` would be more appropriate for these assertions than `expect_equal` unless you need to tolerate some small numeric difference.  (I'm not very consistent about this, either, though.)

  * the logic in these tests feel a bit too close to the logic in the code.  Rather than repeating the logic and risking some of the same bugs in both places, I'd prefer dumber test code that's more straightforward (even if repetitive).  For example, instead of looping and doing complicated parsing, read in the model path for each model separate, assert that a couple of expected lines are there and maybe verify that the expect number of lines match "MAXEVAL|NITER".

Just my two cents.  Again, the tests look fine to me overall.

Reply:
Made the change to `expect_identical`. I take your point on feeling similar to the function code. Thought about that as well, but feel the test expectations (names and values) cover any issues that could come up (in terms of test quality). I mainly like that you get a nice neat object before the tests, that you can inspect if any issues were to arise from the function misbehaving.

Has to be:
```r
> max_evals
[[1]]
  METHOD=1, MAXEVAL METHOD=BAYES, NITER 
                100                 100 

[[2]]
  METHOD=1, MAXEVAL METHOD=BAYES, NITER 
                100                 100 
```
which presents & confirms A) the method was preserved, B) `MAXEVAL`/`NITER` was preserved, and C) the values were correctly changed. I could remove the looping part, but Id still have to parse the files to find those lines and `gsub` the number from them, so I feel the majority of the code would still look similar to what it is now, just without the loop.